fdbclient/BackupContainerFileSystem.actor.cpp

```diff
@@ -637,6 +724,49 @@ class BackupContainerFileSystemImpl {
 				    (desc.contiguousLogEnd.get() - 1) > desc.maxRestorableVersion.get())
 					desc.maxRestorableVersion = desc.contiguousLogEnd.get() - 1;
 			}
+
+			// If there is logs gap after contiguousLogEnd and if current snapshot is restorable(have continuous logs)
+			if (desc.contiguousLogEnd.present() && desc.contiguousLogEnd.get() < s.beginVersion && s.restorable.get()) {
+				if (desc.minRestorableVersion.present() && desc.maxRestorableVersion.present()) {
+					ASSERT(desc.minRestorableVersion.get() < s.beginVersion);
+
+					// check if we have contiguous logs from minRestorableVersion to current snapshot endVersion
+					bool contiguousLogs = false;
+					if (desc.partitioned)
+						contiguousLogs =
+						    isPartitionedLogsContinuous(logs, desc.minRestorableVersion.get(), s.endVersion);
+					else
+						contiguousLogs =
+						    hasContinuousLogsForSnapshot(logs, desc.minRestorableVersion.get(), s.endVersion);
+
+					if (contiguousLogs) {
+						// The previous restorable version can be extended to current snapshot vesion,
+						// so minRestorableVersion remain same
+						desc.maxRestorableVersion = s.endVersion;
+					} else {
+						// Previous restorable version cannot be extended to current snapshot version,
+						// means some logs are missing inbetween.
+						// So set the snapshot beginversion as minRestorableVersion.
+						desc.minRestorableVersion = s.endVersion;
+					}
+				} else {
+					// There is no previous snapshot that is restorable.
+					// Since the current snapshot is restorable, set the snapshot beginversion as minRestorableVersion.
+					desc.minRestorableVersion = s.endVersion;
+				}
+
+				// Find the continuousLogEnd after snapshotEndVersion and set it as
+				// maxRestorableVersion.
+
+				if (desc.partitioned) {
+					// TO DO: Yet to implement similar function findContinuousLogEnd for partitioned logs.
```

Comment:
This is required with the new backup feature which will be enabled only in 7.4. Will implement it in a new PR in 7.4 as I don't want to block the current PR.